packages/runtime/src/widgets/Form/Radio.tsx

```diff
@@ -38,19 +75,60 @@ export const RadioWidget: React.FC<RadioWidgetProps> = (props) => {
     [action],
   );
 
-  const formInstance = Form.useFormInstance();
+  // extract template data
+  const { namedData } = useTemplateData({
+    data: itemTemplate?.data,
+    name: itemTemplate?.name,
+  });
 
-  useEffect(() => {
-    setValue(values?.initialValue?.toString());
-  }, [values?.initialValue]);
+  // handle radio items
+  const radioItems = useMemo(() => {
+    const radioOptions: RadioOptionsProps[] = [];
 
-  useEffect(() => {
-    if (formInstance) {
-      formInstance.setFieldsValue({
-        [values?.id ?? values?.label ?? ""]: value,
+    const getStyle = (styles: EnsembleWidgetStyles): object => ({
+      ...styles,
+      ...(styles?.visible === false ? { display: "none" } : undefined),
+    });
+
+    if (values?.items) {
+      values.items.forEach((item) => {
+        radioOptions.push({
+          children: item.label,
+          disabled: values.enabled === false || item.enabled === false,
+          value: String(item.value),
+          style: getStyle(item.styles as object),
+        });
       });
     }
-  }, [value, formInstance]);
+
+    if (isObject(itemTemplate) && !isEmpty(namedData)) {
+      namedData.forEach((item: unknown) => {
+        const typedItem = get(item, itemTemplate.name) as CustomScope;
+        const evaluateValue = get(
+          item,
+          itemTemplate.value
+            .toString()
+            // eslint-disable-next-line prefer-named-capture-group
+            .replace(/['"]?\$\{([^}]*)\}['"]?/g, "$1"),
+        ) as string | number;
```

Comment:
Again this is forcing value as a string?

Reply:
@evshi here, `itemTemplate.value` will be always string, for example,

```yaml
item-template:
    data: getData.body.users.slice(0, 4)
    name: user
    value: ${user.email}
    template:
      Row:
        styles:
          gap: 10px
          paddingBlock: 10px
          alignItems: center
        children:
          - Image:
              width: 25
              height: 25
              source: ${user.image}
          - Text:
              text: ${user.firstName} ${user.lastName}
```

where user.email will be always string and indicate that which value need to assign that iterate, here it says that use `user.email` as a value.

Reply:
Oh I misread this. You shouldn't need to find the evaluated value - `useItemTemplate` does that for you in the named objects.

Reply:
@sagardspeed2 let's extend useTemplateData to handle an optional `value` prop from itemTemplate so it evaluates for you.